XLS-0033d-multi-purpose-tokens/README.md

```diff
@@ -983,3 +1005,68 @@ As can be seen from the following size comparison table, Trustlines take up appr
+│ │  "Not XRP" bit  ││  Sign bit (always 1;   ││"IsMPT" bit ││                  (61 bits)                  │ │
+│ │0=XRP/MPT; 1=IOU ││       positive)        ││0=XRP; 1=MPT││                                             │ │
+│ └─────────────────┘└────────────────────────┘└────────────┘└─────────────────────────────────────────────┘ │
+└────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+#### Encoding for MPT Values (backward compatible)
+
+This encoding focuses on the first 3 bytes of a MPT:
+
+```
+┌────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ ┌─────────────────┐┌────────────────────────┐┌────────────┐┌─────────────────────────────────────────────┐ │
+│ │        0        ││           1            ││     1      ││                                             │ │
+│ │                 ││                        ││            ││             Remaining MPT Bytes             │ │
+│ │  "Not XRP" bit  ││  Sign bit (always 1;   ││"IsMPT" bit ││                 (389 bits)                  │ │
+│ │0=XRP/MPT; 1=IOU ││       positive)        ││0=XRP; 1=MPT││                                             │ │
+│ └─────────────────┘└────────────────────────┘└────────────┘└─────────────────────────────────────────────┘ │
+└────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+This encoding focuses on the rest of the bytes of a MPT (264 bits):
+
+```
+┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ ┌─────────────────┐┌──────────────────┐┌──────────────────────┐┌──────────────────────────┐┌──────────────────────────┐│
+│ │    [0][1][1]    ││                  ││                      ││                          ││                          ││
+│ │                 ││     Reserved     ││   MPT Amount Value   ││      Sequence            ││     Issuer AcountID      ││
+│ │   IsMPT=true    ││     (5 bits)     ││      (64 bits)       ││      (32 bits)           ││        (160 bits)        ││
+│ │                 ││                  ││                      ││                          ││                          ││
+│ └─────────────────┘└──────────────────┘└──────────────────────┘└──────────────────────────┘└──────────────────────────┘│
+└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+Note: MPT introduces an extra leading byte in front of the MPT value. However, despite the MPT value being 64-bit, only 63 bits can be used. This limitation arises because internally, rippled needs to convert the value to `int64`, which has a smaller positive range compared to `uint64`.
+
+*Question*: Could the ledger perform arithmetics in `uint64`? If not, we would need to decrease the range of the MPT value.
```

Comment:
```suggestion
```

I think this question is answered. We should transition this to an FAQ answer instead.